components/prism-ini.js

```diff
@@ -1,11 +1,37 @@
 Prism.languages.ini= {
-	'comment': /^[ \t]*[;#].*$/m,
-	'selector': /^[ \t]*\[.*?\]/m,
-	'constant': /^[ \t]*[^\s=]+?(?=[ \t]*=)/m,
-	'attr-value': {
-		pattern: /=.*/,
+	'comment': {
+		pattern: /(^[ \f\t\v]*)[#;][^\n\r]*/m,
+		lookbehind: true
+	},
+	'header': {
+		pattern: /(^[ \f\t\v]*)\[[^\n\r\]]*\]?/m,
+		lookbehind: true,
 		inside: {
-			'punctuation': /^[=]/
+			'section-name': {
+				pattern: /(\[[ \f\t\v]*)[^ \f\n\r\t\v\]]+(?:[ \f\t\v]+[^ \f\n\r\t\v\]]+)*/,
+				lookbehind: true,
+				alias: 'selector'
+			},
+			rest: {
+				'punctuation': /\[|\]/
+			}
 		}
-	}
+	},
+	'key': {
+		pattern: /(^[ \f\t\v]*)[^ \f\n\r\t\v=]+(?:[ \f\t\v]+[^ \f\n\r\t\v=]+)*(?=[ \f\t\v]*=)/m,
+		lookbehind: true,
+		alias: 'attr-name'
+	},
+	'value': {
+		pattern: /(=[ \f\t\v]*)[^ \f\n\r\t\v]+(?:[ \f\t\v]+[^ \f\n\r\t\v]+)*/,
+		lookbehind: true,
+		alias: 'attr-value',
+		inside: {
+			'inner-value': {
+				pattern: /("|').+(?=\1)/,
```

Comment:
I thought that it was only recognized as a "string" if it started and ended with quotes (of the same kind). Is that also the case if it's in the middle? E.g. `foo=b"a"r`

Reply:
The parsers I’ve checked treat quotation marks in `"b"a"r"`, `"bar`, `b"a"r`, `ba"r`, `bar"` in almost the same way. The only difference is that Python doesn’t truncate outer _paired_ quotation marks.

Quotes are used to create values with leading and/or trailing spaces.

Parsers consistently strip off leading and trailing spaces. A quotation mark (or any other printable character) before leading spaces or after trailing spaces turns those spaces into internal ones. **These quotes do not have to be paired**: `"␣bar`, `bar␣"`. Of course, the quotation marks affect only adjacent spaces. There can be as many quotes as you like.

The only difference between parsers is that Win32 API and PHP truncate _outer paired_ quotation marks.